service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -268,6 +264,40 @@ public boolean updateWorkspace(
     return updated;
   }
 
+  /** Update a workspace properties */
+  @WriteTransaction
+  public boolean updateWorkspaceProperties(UUID workspaceUuid, Map<String, String> propertyMap) {
+    // get current property in this workspace id
+    String selectPropertiesSql = "SELECT properties FROM workspace WHERE workspace_id = :id";
+    MapSqlParameterSource propertiesParams =
+        new MapSqlParameterSource().addValue("id", workspaceUuid.toString());
+    int rowsAffected;
+    try {
+      String result =
+          jdbcTemplate.queryForObject(selectPropertiesSql, propertiesParams, String.class);
+      logger.info("Retrieved workspace record {}", result);
+      Map<String, String> properties = DbSerDes.jsonToProperties(result);
```

Comment:
can you move L279-289 outside the try catch. In general, we want the try catch scope tight.

Reply:
good catch. I will also correct this in PF-1792